File: Bank_Management_System.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <iomanip>
using namespace std;

// ================= Account Base Class =================
class Account {
protected:
    int accountNumber;
    string name;
    double balance;

public:
    Account() : accountNumber(0), balance(0.0) {}

    virtual void createAccount() {
        cout << "Enter Account Number: ";
        cin >> accountNumber;
        cout << "Enter Name: ";
        cin.ignore();
        getline(cin, name);
        cout << "Enter Initial Balance: ";
        cin >> balance;
    }

    virtual void displayAccount() const {
        cout << "Account Number: " << accountNumber << endl;
        cout << "Name: " << name << endl;
        cout << "Balance: " << balance << endl;
    }

    int getAccountNumber() const { return accountNumber; }
    double getBalance() const { return balance; }
    string getName() const { return name; }

    void deposit(double amount) {
        balance += amount;
        cout << "Deposited " << amount << " successfully.\n";
    }

    bool withdraw(double amount) {
        if (amount > balance) {
            cout << "Insufficient balance!\n";
            return false;
        }
        balance -= amount;
        cout << "Withdrew " << amount << " successfully.\n";
        return true;
    }

    virtual void writeToFile(ofstream &out) const {
        out << accountNumber << "," << name << "," << balance << endl;
    }
};

// ================= Derived Class: SavingsAccount =================
class SavingsAccount : public Account {
public:
    void createAccount() override {
        cout << "Creating Savings Account...\n";
        Account::createAccount();
    }
};

// ================= Bank Management =================
class Bank {
public:
    void createAccount() {
        SavingsAccount acc;
        acc.createAccount();
        ofstream out("accounts.txt", ios::app);
        acc.writeToFile(out);
        out.close();
        cout << "Account created and saved successfully!\n";
    }

    void depositMoney() {
        int accNo;
        double amount;
        cout << "Enter Account Number: ";
        cin >> accNo;
        cout << "Enter Amount to Deposit: ";
        cin >> amount;

        updateBalance(accNo, amount, true);
    }

    void withdrawMoney() {
        int accNo;
        double amount;
        cout << "Enter Account Number: ";
        cin >> accNo;
        cout << "Enter Amount to Withdraw: ";
        cin >> amount;

        updateBalance(accNo, amount, false);
    }

    void accountStatement() {
        int accNo;
        cout << "Enter Account Number: ";
        cin >> accNo;

        ifstream in("accounts.txt");
        string line;
        bool found = false;
        while (getline(in, line)) {
            int aNo;
            string aName;
            double aBal;

            size_t pos1 = line.find(",");
            size_t pos2 = line.find_last_of(",");

            aNo = stoi(line.substr(0, pos1));
            aName = line.substr(pos1 + 1, pos2 - pos1 - 1);
            aBal = stod(line.substr(pos2 + 1));

            if (aNo == accNo) {
                cout << "=== Account Statement ===\n";
                cout << "Account Number: " << aNo << "\n";
                cout << "Name: " << aName << "\n";
                cout << "Balance: " << aBal << "\n";
                found = true;
                break;
            }
        }
        in.close();
        if (!found) cout << "Account not found!\n";
    }

private:
    void updateBalance(int accNo, double amount, bool isDeposit) {
        ifstream in("accounts.txt");
        ofstream temp("temp.txt");

        string line;
        bool found = false;

        while (getline(in, line)) {
            int aNo;
            string aName;
            double aBal;

            size_t pos1 = line.find(",");
            size_t pos2 = line.find_last_of(",");

            aNo = stoi(line.substr(0, pos1));
            aName = line.substr(pos1 + 1, pos2 - pos1 - 1);
            aBal = stod(line.substr(pos2 + 1));

            if (aNo == accNo) {
                if (isDeposit) {
                    aBal += amount;
                    cout << "Deposited successfully. New Balance: " << aBal << endl;
                } else {
                    if (amount > aBal) {
                        cout << "Insufficient balance!\n";
                    } else {
                        aBal -= amount;
                        cout << "Withdrawal successful. New Balance: " << aBal << endl;
                    }
                }
                found = true;
            }
            temp << aNo << "," << aName << "," << aBal << endl;
        }

        in.close();
        temp.close();

        remove("accounts.txt");
        rename("temp.txt", "accounts.txt");

        if (!found) cout << "Account not found!\n";
    }
};

// ================= Main Menu =================
int main() {
    Bank bank;
    int choice;

    do {
        cout << "\n=== Bank Management System ===\n";
        cout << "1. Create Account\n";
        cout << "2. Deposit Money\n";
        cout << "3. Withdraw Money\n";
        cout << "4. Account Statement\n";
        cout << "5. Exit\n";
        cout << "Enter choice: ";
        cin >> choice;

        switch (choice) {
            case 1: bank.createAccount(); break;
            case 2: bank.depositMoney(); break;
            case 3: bank.withdrawMoney(); break;
            case 4: bank.accountStatement(); break;
            case 5: cout << "Exiting...\n"; break;
            default: cout << "Invalid choice!\n";
        }
    } while (choice != 5);

    return 0;
}

File: Bank_Management_System.py
import sys

class BankAccount:
    def __init__(self, acc_number, name, balance=0):
        self.acc_number = acc_number
        self.name = name
        self.balance = balance

    def display(self):
        print(f"\nAccount Number: {self.acc_number}")
        print(f"Account Holder: {self.name}")
        print(f"Balance: ${self.balance:.2f}")

    def deposit(self, amount):
        if amount > 0:
            self.balance += amount
            print(f"\n${amount:.2f} deposited successfully.")
        else:
            print("Invalid deposit amount.")

    def withdraw(self, amount):
        if amount <= self.balance:
            self.balance -= amount
            print(f"\n${amount:.2f} withdrawn successfully.")
        else:
            print("Insufficient balance.")

# Dictionary to hold accounts
accounts = {}

def create_account():
    acc_number = input("Enter new account number: ")
    if acc_number in accounts:
        print("Account already exists.")
        return
    name = input("Enter account holder name: ")
    initial_deposit = float(input("Enter initial deposit: "))
    accounts[acc_number] = BankAccount(acc_number, name, initial_deposit)
    print("Account created successfully!")

def view_account():
    acc_number = input("Enter account number: ")
    if acc_number in accounts:
        accounts[acc_number].display()
    else:
        print("Account not found.")

def deposit_money():
    acc_number = input("Enter account number: ")
    if acc_number in accounts:
        amount = float(input("Enter amount to deposit: "))
        accounts[acc_number].deposit(amount)
    else:
        print("Account not found.")

def withdraw_money():
    acc_number = input("Enter account number: ")
    if acc_number in accounts:
        amount = float(input("Enter amount to withdraw: "))
        accounts[acc_number].withdraw(amount)
    else:
        print("Account not found.")

def main():
    while True:
        print("\n==== Bank Management System ====")
        print("1. Create Account")
        print("2. View Account")
        print("3. Deposit Money")
        print("4. Withdraw Money")
        print("5. Exit")

        choice = input("Enter your choice (1-5): ")

        if choice == '1':
            create_account()
        elif choice == '2':
            view_account()
        elif choice == '3':
            deposit_money()
        elif choice == '4':
            withdraw_money()
        elif choice == '5':
            print("Thank you for using the Bank Management System.")
            sys.exit()
        else:
            print("Invalid choice. Try again.")

if __name__ == "__main__":
    main()
